src/libraries/System.Net.Http/src/System/Net/Http/MultipartContent.cs

```diff
@@ -669,6 +681,31 @@ private static void ValidateReadArgs(byte[] buffer, int offset, int count)
             public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken) { throw new NotSupportedException(); }
             public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken = default) { throw new NotSupportedException(); }
         }
+
+
+        private static void WriteToStream(Stream stream, string content, Encoding encoding)
+        {
+            if (ReferenceEquals(encoding, Encoding.Latin1) ? content.Length <= 512
+                : (ReferenceEquals(encoding, Encoding.UTF8) && content.Length <= 128))
+            {
+                // Fast-path for the expected common cases
+                Span<byte> buffer = stackalloc byte[512];
+                int written = encoding.GetBytes(content, buffer);
+                stream.Write(buffer.Slice(0, written));
+            }
+            else
+            {
+                stream.Write(encoding.GetBytes(content));
+            }
+        }
+
+        private static void WriteLatin1ToStream(Stream stream, string content)
+        {
+            Debug.Assert(ReferenceEquals(Encoding.Latin1, HttpRuleParser.DefaultHttpEncoding));
+
+            WriteToStream(stream, content, HttpRuleParser.DefaultHttpEncoding);
```

Comment:
Why can't we use this exact line instead of `WriteLatin1ToStream`? Is this helper necessary if it's just a one-liner?

Reply:
I renamed it to just be an overload of WriteToStream

```c#
private static void WriteToStream(Stream stream, string content) =>
    WriteToStream(stream, content, HttpRuleParser.DefaultHttpEncoding);
```

So that code like this doesn't have to be too verbose in passing in the encoding every time
```c#
WriteToStream(stream, CrLf + "--"); // const strings
WriteToStream(stream, _boundary);
WriteToStream(stream, CrLf);
```